python/dgl/convert.py

```diff
@@ -1688,36 +1793,54 @@ def to_networkx(g, node_attrs=None, edge_attrs=None):
     >>> nx_g.edges(data=True)
     OutMultiEdgeDataView([(1, 1, {'id': 0, 'h1': tensor([1.]), 'h2': tensor([0., 0.])}),
                           (2, 3, {'id': 1, 'h1': tensor([1.]), 'h2': tensor([0., 0.])})])
+
+    With a heterogeneous graph:
+
+    >>> g = dgl.heterograph({
+    ...     ('user', 'follows', 'user'): (torch.tensor([0, 1]), torch.tensor([1, 2])),
+    ...     ('user', 'follows', 'topic'): (torch.tensor([1, 1]), torch.tensor([1, 2])),
+    ...     ('user', 'plays', 'game'): (torch.tensor([0, 3]), torch.tensor([3, 4]))
+    ... })
+    ... g.ndata['n'] = {
+    ...     'game': torch.zeros(5, 1),
+    ...     'user': torch.ones(4, 1)
+    ... }
+    ... g.edata['e'] = {
+    ...     ('user', 'follows', 'user'): torch.zeros(2, 1),
+    ...     'plays': torch.ones(2, 1)
+    ... }
+    >>> nx_g = dgl.to_networkx(g, node_attrs=['n'], edge_attrs=['e'])
+    >>> nx_g.nodes(data=True)
+    NodeDataView({0: {'type': 'game', 'n': tensor([0.])},
+                  1: {'type': 'game', 'n': tensor([0.])},
+                  2: {'type': 'game', 'n': tensor([0.])},
+                  3: {'type': 'game', 'n': tensor([0.])},
+                  4: {'type': 'game', 'n': tensor([0.])},
+                  5: {'type': 'topic'},
+                  6: {'type': 'topic'},
+                  7: {'type': 'topic'},
+                  8: {'type': 'user', 'n': tensor([1.])},
+                  9: {'type': 'user', 'n': tensor([1.])},
+                  10: {'type': 'user', 'n': tensor([1.])},
+                  11: {'type': 'user', 'n': tensor([1.])}})
+    >>> nx_g.edges(data=True)
+    OutMultiEdgeDataView([(8, 9, {'id': 2, 'triples': ('user', 'follows', 'user'), 'e': tensor([0.])}),
+                          (8, 3, {'id': 4, 'triples': ('user', 'plays', 'game'), 'e': tensor([1.])}),
+                          (9, 6, {'id': 0, 'triples': ('user', 'follows', 'topic')}),
+                          (9, 7, {'id': 1, 'triples': ('user', 'follows', 'topic')}),
+                          (9, 10, {'id': 3, 'triples': ('user', 'follows', 'user'), 'e': tensor([0.])}),
+                          (11, 4, {'id': 5, 'triples': ('user', 'plays', 'game'), 'e': tensor([1.])})])
     """
     if g.device != F.cpu():
         raise DGLError(
             "Cannot convert a CUDA graph to networkx. Call g.cpu() first."
         )
     if not g.is_homogeneous:
```

Comment:
It might the code more readable to do `if g.is_homogeneous:` instead.

Reply:
I agree. Fixed.